### ISSUE TITLE
Checkpoint management: list and export a specific checkpoint

### ISSUE THREAD

**Issue by will-lamerton** [Nano-Collective/nanotune#61]
## Problem

`saveEvery` (default 50) makes MLX write periodic checkpoints into `.nanotune/adapters/`:

```
0000050_adapters.safetensors
0000100_adapters.safetensors
0000150_adapters.safetensors
adapters.safetensors          <- always the final state
```

But `nanotune export` hardcodes `adapters.safetensors` (`src/commands/export.tsx`). The intermediate checkpoints are written and then never used by anything.

This matters most in the case they exist for: if a model overfits, the best checkpoint is an earlier one, and there is currently no way to export it without renaming files by hand.

## Proposal

- `nanotune checkpoints` (or `nanotune train --list-checkpoints`) listing what's available, with iteration number, timestamp, and size
- `nanotune export --checkpoint 100` to fuse and export a specific checkpoint
- If run history lands (#60), show the val loss at each checkpoint so users can pick the best one rather than guessing

## Acceptance criteria

- [ ] Available checkpoints are discoverable from the CLI
- [ ] Any checkpoint can be exported, not just the final one
- [ ] Selecting a nonexistent checkpoint fails with a clear message listing what is available
- [ ] Docs updated at `docs/commands/export.md`

**Comment by Dhirenderchoudhary** [comment]
@will-lamerton can i go for it and raise PR asap.
